### ISSUE TITLE
`TryWithResourcesFilter` does not work with Java 11+

### ISSUE THREAD

**Issue by Vampire** [hcoles/pitest#689]
With this code:
```java
Properties versionProperties = new Properties();
try (InputStream versionPropertiesStream = versionPropertiesResource.openStream()) {
    versionProperties.load(versionPropertiesStream);
} catch (IOException ignored) {
}
```
in Java 8, 9 and 10, the `TryWithResourcesFilter` properly filters out the mutants.
Starting with Java 11 this is no longer the case:
![grafik](https://user-images.githubusercontent.com/325196/66059272-0d924900-e53c-11e9-82d3-ad7bbe0646ca.png)
![grafik](https://user-images.githubusercontent.com/325196/66059336-2a2e8100-e53c-11e9-9010-136f27c7f92a.png)

**Comment by Vampire** [comment]
Btw. @hcoles you should maybe also checkin the sources for the pre-compiled test classes or additionally test them freshly compiled.
Currently it is hard to add the test cases for Java 11 compiler.
At least unless one has the sources or decompiles the precompiled ones and then recompiles them.

**Comment by Vampire** [comment]
Also tested with 12 and 13, same result

**Comment by Vampire** [comment]
Hm, besides that the sequence of bytecodes seems to have changed slightly, the bytecodes are not split between the `try` and `catch` line, so the current tactic of recognizing the bytecode sequence and ignoring all mutants on that line might not work properly.
The modified sequence (5 instructions less) still matches the catch line number.
But the mutants are not all on the catch line anymore.
But this might also be a different additional problem, as looking at the `opcodeStack` for line 79 in the `TryWithResourcesMethodVisitor`, I only see one `INVOKEVIRTUAL` and one `ASTORE` which should be my `openStream()` call and the assignment I think.
This makes me wonder why the 5 `NO_COVERAGE` mutants are reported for line 79 instead of 81.
I think I'll stop investigation here and let it for you, as you are more familiar with this.

**Comment by hcoles** [comment]
Clsoing as this is being fixed under #766
